pepys_import/file/highlighter/support/line.py

```diff
@@ -66,17 +67,33 @@ def tokens(self, reg_exp=WHITESPACE_DELIM, strip_char=""):
                     char_index = token_str.find(strip_char)
                     if char_index == 0:
                         token_str = token_str[1:]
-                        # and ditch any new whitespace
+                    # and ditch any new whitespace
                     token_str = token_str.strip()
 
                 subtoken = SubToken(match.span(), token_str, int(child.line_start), child.chars)
-
-                # the token object expects an array of SubTokens, as it could be
-                # a composite object
+                # the token object expects an array of SubTokens, as it could be a composite object
                 list_of_subtokens = [subtoken]
-
                 tokens_array.append(Token(list_of_subtokens, self.highlighted_file))
 
+            for match in finditer(quoted_name, child.text):
+                original_name = match.group()
+                quoted_sensor_name = original_name[1:-1].strip()
+                subtoken_sensor_name = SubToken(
+                    match.span(), quoted_sensor_name, int(child.line_start), child.chars
+                )
+                subtokens_sensor_name = [subtoken_sensor_name]
+
+                del_indexes = list()

```

Comment:
Could you add some comments here explaining what we're trying to do? I'm a little bit confused by exactly what the aim of this chunk of code is

Reply:
Sure. I'll add more comments to the code and let you know 👍

Reply:
@robintw , I reliased that the second for loop was really confusing. That's why I moved the quoted text's regex above of the main loop and added many comments:
https://github.com/debrief/pepys-import/pull/411/commits/04b3b18c7a3bb88c7ae586a32748a7849d5c66b5

What I was doing is as follows: Use the main loop to split a line into tokens-> Search for quoted texts in line(the second for loop)-> If there is any, find previously added tokens and delete them in tokens_array. Add the quoted text instead.

But right now, the code does the following: Search for a quoted text in line (use `re.search() `instead of for loop)->Find it's start and end indexes in the line-> In the main loop, check if the quoted text contains token. If it includes, then doesn't add the token to tokens_array.

I hope that the code is tidier now. If you have any other questions, I'd happily answer.